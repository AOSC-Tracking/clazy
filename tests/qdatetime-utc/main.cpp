#include <QtCore/QDateTime>

void test()
{
    QDateTime::currentDateTime(); // OK
    QDateTime::currentDateTimeUtc(); // OK
    QDateTime::currentDateTime().toUTC(); // Warning
#if QT_VERSION_MAJOR == 5
    QDateTime::currentDateTime().toTime_t(); // Warning, but this method is removed in Qt6
#endif
    QDateTime::currentDateTime().toMSecsSinceEpoch(); // Warning
    QDateTime::currentDateTimeUtc().toMSecsSinceEpoch(); // Warning, this may even be improved!
    QDateTime::currentDateTime().toSecsSinceEpoch(); // Warning
    QDateTime::currentDateTimeUtc().toSecsSinceEpoch(); // Warning, this may even be improved!
}
