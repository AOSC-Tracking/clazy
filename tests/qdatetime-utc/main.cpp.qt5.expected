qdatetime-utc/main.cpp:7:5: warning: Use QDateTime::currentDateTimeUtc() instead. It is significantly faster [-Wclazy-qdatetime-utc]
qdatetime-utc/main.cpp:9:5: warning: Use QDateTime::currentMSecsSinceEpoch() instead. It is significantly faster [-Wclazy-qdatetime-utc]
qdatetime-utc/main.cpp:11:5: warning: Use QDateTime::currentMSecsSinceEpoch() instead. It is significantly faster [-Wclazy-qdatetime-utc]
qdatetime-utc/main.cpp:12:5: warning: Use QDateTime::currentMSecsSinceEpoch() instead. It is significantly faster [-Wclazy-qdatetime-utc]
qdatetime-utc/main.cpp:13:5: warning: Use QDateTime::currentSecsSinceEpoch() instead. It is significantly faster [-Wclazy-qdatetime-utc]
qdatetime-utc/main.cpp:14:5: warning: Use QDateTime::currentSecsSinceEpoch() instead. It is significantly faster [-Wclazy-qdatetime-utc]
